api/pkg/instrumentation/metrics/prometheus.go

```diff
@@ -180,3 +180,32 @@ func (p PrometheusClient) RecordGauge(
 
 	return nil
 }
+
+func (p PrometheusClient) RegisterAdditionalMetrics(
+	gaugeMap map[MetricName]PrometheusGaugeVec,
+	histogramMap map[MetricName]PrometheusHistogramVec,
+	counterMap map[MetricName]PrometheusCounterVec,
+) error {
+	for name, obs := range gaugeMap {
+		if _, ok := p.gaugeMap[name]; ok {
+			return errors.Newf("Name of metric already exists: %s", name)
```

Comment:
Instead of "Name of metric", it may be more useful to say "Name of gauge", "Name of histogram", etc.

Or perhaps `Metric of type Gauge with name XXX already exists` ?